src/dmd/cond.d

```diff
@@ -151,13 +153,22 @@ extern (C++) final class StaticForeach : RootObject
         el = el.ctfeInterpret();
         if (el.op == TOK.int64)
         {
-            dinteger_t length = el.toInteger();
-            auto es = new Expressions();
-            foreach (i; 0 .. length)
+            Expressions *es = void;
+            if (auto ale = aggr.isArrayLiteralExp())
             {
-                auto index = new IntegerExp(loc, i, Type.tsize_t);
-                auto value = new IndexExp(aggr.loc, aggr, index);
-                es.push(value);
+                // Directly use the elements of the array for the TupleExp creation
+                es = ale.elements;
+            }
+            else
+            {
+                const length = cast(size_t)el.toInteger();
+                es = new Expressions(length);
+                foreach (i; 0 .. length)
+                {
+                    auto index = new IntegerExp(loc, i, Type.tsize_t);
+                    auto value = new IndexExp(aggr.loc, aggr, index);
+                    (*es)[i] = value;
+                }
```

Comment:
Inserting an `assert(false)` here, it seems that this branch is now dead.  It would be good if a test can be found that hits this, but I guess it's fine, as this code has already been extensively tested from before this change.

Reply:
It's still used for StringExp and NullExp.
The first happens with static `foreach (i; "asdfg") {}`
The second with:
```
static foreach(i; "") {} // or
enum int[] a = null;
static foreach(i; a) {}
```

Reply:
Could those be added to the testsuite then?